source/development/core-dumps-generation.rst

```diff
@@ -0,0 +1,152 @@
+
+  # killall -SEGV wazuh-syscheckd
+
+You will see something like::
+
+  [1]    67868 segmentation fault (core dumped)  ./wazuh-syscheckd
+
+And then a core dump file will be created. In this case, it's ``/cores/core.67868``.
+
+Now you can fire up the ``lldb`` debugger (making sure the working directory contains the ``wazuh-syscheckd`` executable)::
+
+  # lldb
+
+Once the lldb prompt is showing, enter the following command (use the right path for the core dump that was created for
+the ``--core`` path)::
+
+  (lldb) target create wazuh-syscheckd --core /cores/core.67868 
+
+Then you will get see a message similar to this::
+
+  Core file '/cores/core.67868' (x86_64) was loaded.
+
+Now you can run the debugging commands as you would if you were attached to the running process, except, of course, you will not
+be able to step through the code or run code.
+
+Print the backtrace with ``bt``::
+
+  (lldb) bt
+  * thread #1
+    * frame #0: 0x000000010a203034 wazuh-syscheckd`main + 4
+      frame #1: 0x000000010f4fd51e dyld`start + 462
+
+Switch to thread 1 using ``t 1`` (the source code shown here is just an example of a real intentional segmentation fault)::
+
+  (lldb) t 1
+  * thread #1
+      frame #0: 0x0000000108ef6bba wazuh-syscheckd`main(argc=1, argv=0x00007ff7b7011aa8) at main.c:52:8 [opt]
+    49       int start_realtime = 0;
+    50  
+    51       int *a = 0x0423;
+  -> 52       *a = 0;
+    53  
+    54       /* Set the name */
+    55       OS_SetName(ARGV0);
+
+Print the value of local variables with ``frame variable`` (some variables might be missing depending on the optimization
+level used when compiling)::
+
+  (lldb) frame variable
+  (int) argc = 1
+  (char **) argv = 0x00007ff7b7011aa8
+  (int) start_realtime = 0
+  (directory_t *) dir_it = NULL
+  (int) run_foreground = 0
+  (int) test_config = 0
+  (int) debug_level = 0
+  (char *) home_path = <variable not available>
+  (const char *) cfg = <variable not available>
+  (int) c = <variable not available>
+  (gid_t) gid = <variable not available>
+  (int) r = <variable not available>
+  (const char *) group = <no location, value may have been optimized out>
+  (int *) a = <no location, value may have been optimized out>
+
+Note about debugging symbols
+----------------------------
+
+Debugging symbols are created separately by default and are available to download at our :ref:`debug packages list<macos-dbg-symbols-packages>`. This means binary files (executables and .dylib shared libraries)
+have no debugging symbols in them and .dSYM bundle folders are created inside the ``src/symbols`` directory.
+
+For example, after compiling, you get ``src/wazuh-syscheckd`` and ``src/symbols/wazuh-syscheckd.dSYM``. The ``lldb``
+debugger should automatically find the matching dSYM bundle, as long as it is findable by macOS's Spotlight [2]_. However, the path to the dSYM bundle can be manually specified by using the ``add-dsym``::
+
+  add-dsym <path to dSYM bundle>
```

Comment:
This will be more helpful if we basically said that, in case spotlight does not reach budles, they could be pointed explicitly by add-syms, whether they are download from wazuh or compiled from sources